pkg/repo/index_test.go

```diff
@@ -214,6 +214,42 @@ func TestMerge(t *testing.T) {
 
 }
 
+func TestMergePrerelease(t *testing.T) {
+	ind1 := NewIndexFile()
+	ind1.Add(&chart.Metadata{
+		Name:    "dreadnought",
+		Version: "0.2.0",
+	}, "dreadnought-0.2.0.tgz", "http://example.com", "aaaa")
+
+	ind2 := NewIndexFile()
+	ind2.Add(&chart.Metadata{
+		Name:    "dreadnought",
+		Version: "0.1.0",
+	}, "dreadnought-0.1.0.tgz", "http://example.com", "aaaabbbb")
+	ind2.Add(&chart.Metadata{
+		Name:    "dreadnought",
+		Version: "0.1-dev",
+	}, "dreadnought-0.1-dev.tgz", "http://example.com", "ccccbbbb")
+
+	ind1.Merge(ind2)
+
+	if len(ind1.Entries) != 1 {
+		t.Errorf("Expected 1 entries, got %d", len(ind1.Entries))
+	}
+	vs := ind1.Entries["dreadnought"]
+	if len(vs) != 3 {
+		t.Errorf("Expected 3 versions, got %d", len(vs))
```

Comment:
Can you also check for the case where the exact same version is in both and it is overwritten via the merge. This would have caught the the error with the use of `Has` and be good for future proofing.

Reply:
Do you mean an exact same version but with a different digest for instance? Can do.

Reply:
I mean where index 1 and index 2 have the same version... the output index should have only one of them and it should be the right one.

Reply:
The part that confused me is if they're the same version, how do you know which one is the "right" one?

Reply:
Looks like there's some existing logic that assumes merges will not overwrite existing contents, so I made the test follow that same approach.